cpp/include/raft/cudart_utils.h

```diff
@@ -267,12 +267,12 @@ void allocate(Type*& ptr, size_t len, bool setZero = false) {
 }
 
 /** helper method to get max usable shared mem per block parameter */
-inline int getSharedMemPerBlock() {
+__host__ __device__ inline int getSharedMemPerBlock() {
   int devId;
-  CUDA_CHECK(cudaGetDevice(&devId));
+  cudaGetDevice(&devId);
```

Comment:
Any reason we can't query this on host and pass it in? That would allow us to keep the checks here and not need to query it for each individual block.

Reply:
Ya, I didn't make this change either. If you remember, I had asked you a couple of weeks ago about this as well. Happy to revert it to just a host function

Reply:
Interesting, yeah I don't recall making this change either. I suppose it's possible I could have made it early on when I was trying something out, but I had tried reverting tthe change at one point and it broke something that I didn't believe I had written. No biggie either way, but it would probably be best to just revert this if we can.